File: frontend/src/types/loginHistory.ts
/**
 * 登录历史相关类型定义
 */

// 登录状态
export type LoginStatus = 'success' | 'failed' | 'locked'

// 登录历史记录
export interface LoginHistoryRecord {
  id: number
  user_id: number
  username?: string
  ip_address?: string
  user_agent?: string
  location?: string
  login_status: LoginStatus
  failure_reason?: string
  login_at: string
  created_at: string
}

// 登录历史列表响应
export interface LoginHistoryListResponse {
  items: LoginHistoryRecord[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

// 登录历史查询参数
export interface LoginHistoryParams {
  page?: number
  page_size?: number
  status?: LoginStatus
  user_id?: number
}

// 登录错误详情（从后端返回）
export interface LoginErrorDetail {
  message: string
  code: 'wrong_password' | 'account_locked' | 'account_disabled'
  remaining_attempts?: number
  remaining_seconds?: number
  locked_until?: string
}

File: backend/app/schemas/login_history.py
"""
登录历史相关 Schemas
"""
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class LoginHistoryResponse(BaseModel):
    """登录历史记录响应"""

    id: int
    user_id: int
    username: Optional[str] = None
    ip_address: Optional[str] = None
    user_agent: Optional[str] = None
    location: Optional[str] = None
    login_status: str = Field(..., description="登录状态: success/failed/locked")
    failure_reason: Optional[str] = None
    login_at: datetime
    created_at: datetime

    class Config:
        from_attributes = True


class LoginHistoryListResponse(BaseModel):
    """登录历史列表响应"""

    items: list[LoginHistoryResponse]
    total: int
    page: int
    page_size: int
    total_pages: int
